apps/hosts/views.py

```diff
@@ -254,6 +254,47 @@ def remove_self_from_challenge_host_team(request, challenge_host_team_pk):
         return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
 
 
+@api_view(["DELETE"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def delete_challenge_host_from_team(request, challenge_host_team_pk, challenge_host_pk):
+    """
+    Deletes a challenge_host from a challenge_host_Team
+    """
+    try:
+        challenge_host_team = ChallengeHostTeam.objects.get(pk=challenge_host_team_pk)
+    except ChallengeHostTeam.DoesNotExist:
+        response_data = {"error": "ChallengeHostTeam does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    try:
+        challenge_host = ChallengeHost.objects.get(pk=challenge_host_pk)
+    except ChallengeHost.DoesNotExist:
+        response_data = {"error": "ChallengeHost does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    if challenge_host_team.created_by == request.user:
+
+        if (
+            challenge_host.user == request.user
+        ):  # when the user tries to remove himself
+            response_data = {
+                "error": "You are not allowed to remove yourself since you are admin. Please delete the team if you want to do so!"
+            }  # noqa: ignore=E501
+            return Response(
+                response_data, status=status.HTTP_406_NOT_ACCEPTABLE
+            )
+        else:
+            challenge_host.delete()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+    else:
+        response_data = {
+            "error": "Sorry, you do not have permissions to remove this challenge host"
+        }
+        return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
```

Comment:
Similar functionality is implemented here: https://github.com/Cloud-CV/EvalAI/blob/master/apps/hosts/views.py#L204-L206 
It also has URL assigned to it. Could you implement this there itself?
The linked one doesn't check the permissions of the user before deleting. I might be missing something, but as I understand the code, that should be a bug. Implementing these checks there would be helpful.

Please let me know if I am missing something.

Reply:
> Similar functionality is implemented here: https://github.com/Cloud-CV/EvalAI/blob/master/apps/hosts/views.py#L204-L206
> It also has URL assigned to it. Could you implement this there itself?
> The linked one doesn't check the permissions of the user before deleting. I might be missing something, but as I understand the code, that should be a bug. Implementing these checks there would be helpful.
> 
> Please let me know if I am missing something.

@nikochiko That should have been the code for deleting host member but when i used the url associated with that particular function, it let me to delete the whole team as it is not performing any kind of check. If that is not written for any other purpose i'll edit that only to get all the checks done and then perform the deletion. Please let me know if can do it.

Reply:
It is probably a bug in that case.
Go ahead and add the changes there itself instead of having two URLs/views for the same task.
For keeping record, you can create an issue stating the bug and attaching a GIF.
cc: @RishabhJain2018
reference: https://github.com/Cloud-CV/EvalAI/blob/master/apps/hosts/views.py#L204-L206